domain/src/main/java/org/oppia/android/domain/classroom/ClassroomController.kt

```diff
@@ -0,0 +1,386 @@
+        assetName = "classrooms",
+        baseMessage = ClassroomIdList.getDefaultInstance()
+      )
+      return ClassroomList.newBuilder().apply {
+        addAllClassroomSummary(
+          classroomIdList.classroomIdsList.map { classroomId ->
+            createEphemeralClassroomSummary(classroomId, contentLocale)
+          }.filter { ephemeralClassroomSummary ->
+            ephemeralClassroomSummary.classroomSummary.topicSummaryList.any { topicSummary ->
+              topicSummary.topicPlayAvailability.availabilityCase == AVAILABLE_TO_PLAY_NOW
+            }
+          }
+        )
+      }.build()
+    } else loadClassroomListFromJson(contentLocale)
+  }
+
+  private fun loadClassroomListFromJson(contentLocale: OppiaLocale.ContentLocale): ClassroomList {
+    val classroomIdJsonArray = jsonAssetRetriever
+      .loadJsonFromAsset("classrooms.json")
+      ?.getJSONArray("classroom_id_list")
+      ?: return ClassroomList.getDefaultInstance()
+    val classroomListBuilder = ClassroomList.newBuilder()
+    for (i in 0 until classroomIdJsonArray.length()) {
+      val classroomId = classroomIdJsonArray.optString(i)
+      val ephemeralClassroomSummary = createEphemeralClassroomSummary(classroomId, contentLocale)
+      val hasPublishedTopics =
+        ephemeralClassroomSummary.classroomSummary.topicSummaryList.any { topicSummary ->
+          topicSummary.topicPlayAvailability.availabilityCase == AVAILABLE_TO_PLAY_NOW
+        }
+      if (hasPublishedTopics) classroomListBuilder.addClassroomSummary(ephemeralClassroomSummary)
+    }
+    return classroomListBuilder.build()
+  }
+
+  private fun createEphemeralClassroomSummary(
+    classroomId: String,
+    contentLocale: OppiaLocale.ContentLocale
+  ): EphemeralClassroomSummary {
+    return EphemeralClassroomSummary.newBuilder().apply {
+      classroomSummary = createClassroomSummary(classroomId)
+      writtenTranslationContext =
+        translationController.computeWrittenTranslationContext(
+          classroomSummary.writtenTranslationsMap, contentLocale
+        )
+    }.build()
+  }
+
+  private fun createClassroomSummary(classroomId: String): ClassroomSummary {
+    return if (loadLessonProtosFromAssets) {
+      val classroomRecord = assetRepository.loadProtoFromLocalAssets(
+        assetName = classroomId,
+        baseMessage = ClassroomRecord.getDefaultInstance()
+      )
+      return ClassroomSummary.newBuilder().apply {
+        this.classroomId = classroomId
+        putAllWrittenTranslations(classroomRecord.writtenTranslationsMap)
+        classroomTitle = classroomRecord.translatableTitle
+        classroomThumbnail = createClassroomThumbnailFromProto(
+          classroomId,
+          classroomRecord.classroomThumbnail
+        )
+        addAllTopicSummary(
+          classroomRecord.topicPrerequisitesMap.keys.toList().map { topicId ->
+            createTopicSummary(topicId, classroomId)
+          }
+        )
+      }.build()
```

Comment:
Looks like this and `getTopicIdListFromClassroomRecord` both create a list of TopicSummary. I am wpnderiing if one of the models could do without the object.

In the UI PR, you are fetching both the classromlist and topiclist seperately, yet, ideally, if you have the list of classrooms, you can subsequently extract the list of topics from it.

Reply:
> Looks like this and `getTopicIdListFromClassroomRecord` both create a list of TopicSummary. I am wpnderiing if one of the models could do without the object.

Removed fetching of `TopicSummary` from `getTopicIdListFromClassroomRecord` function as only `TopicId`s are required here.

> In the UI PR, you are fetching both the classromlist and topiclist seperately, yet, ideally, if you have the list of classrooms, you can subsequently extract the list of topics from it.

This is a good point, maybe we could discuss this further in the UI PR.